python/lsst/meas/algorithms/reinterpolate_pixels.py

```diff
@@ -0,0 +1,140 @@
+# This file is part of meas_algorithms.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("ReinterpolatePixelsConfig", "ReinterpolatePixelsTask")
+
+
+import math
+
+import lsst.afw.detection as afwDetection
+import lsst.afw.math as afwMath
+import lsst.meas.algorithms as measAlg
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.afw.image import MaskedImageF
+
+
+class ReinterpolatePixelsConfig(pexConfig.Config):
+    """Config for ReinterpolatePixelsTask"""
+
+    kernelFwhm = pexConfig.Field[float](
+        doc="FWHM of double Gaussian smoothing kernel.",
+        default=1.0,
+    )
+
+    growFootprints = pexConfig.Field[int](
+        doc="Number of pixels to grow footprints for pixels.",
+        default=1,
+    )
+
+    maskNameList = pexConfig.ListField[str](
+        doc="Mask plane name.",
+        default=["SAT"],
+    )
+
+    fallbackValue = pexConfig.Field[float](
+        doc="Value of last resort for interpolation.",
+        default=None,
+        optional=True,
+    )
+
+
+class ReinterpolatePixelsTask(pipeBase.Task):
+    """Task to reinterpolate pixels"""
+
+    ConfigClass = ReinterpolatePixelsConfig
+    _DefaultName = "reinterpolatePixels"
+
+    def run(self, exposure):
+        """Reinterpolate pixels over ``exposure``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to interpolate over.
+
+        Notes
+        -----
+        ``exposure`` is modified in place and will become the reinterpolated exposure.
+        """
+        mask = exposure.mask
+
+        if self.config.growFootprints > 0 and "SAT" in self.config.maskNameList:
+            # If we are interpolating over an area larger than the original
+            # masked region, we need to expand the original mask bit to the
+            # full area to explain why we interpolated there.
+            from lsst.ip.isr import growMasks
+
+            growMasks(mask, radius=self.config.growFootprints, maskNameList=["SAT"], maskValue="SAT")
+
+        from lsst.ip.isr import Defects
```

Comment:
I hadn't realized the production code (not just the tests) depended on ip_isr; I think this is something we need to fix.

Given how thin the `Defects.fromFootprintList` method is, you should just copy its implementation here.

Reply:
I noticed that `Defects.fromFootprintList` returns an instance of the `lsst.ip.isr.Defects` class which belongs to ip_isr. Doesn't copying the implementation of `Defects.fromFootprintList` here mean that I need to remake an entire `lsst.ip.isr.Defects` class here? I'm unsure if this aligns with our intentions.

Reply:
Instead of making a `Defects` instance, just copy the bit that makes an iterable of ``lsst.meas.algorithms.Defect`, since that iterable is also what you need to pass to `interpolateOverDefects`.

Reply:
It worked, but I also needed to map the `lsst.meas.algorithms.Defect` function to each `lsst.afw.detection.footprintToBBoxList(fp)` for `fp` in `fpList` to get `List[lsst.meas.algorithms.Defect]`.